virtualenv.py

```diff
@@ -459,19 +465,35 @@ def _find_file(filename, folders):
     return False, filename
 
 
-def file_search_dirs():
-    here = os.path.dirname(os.path.abspath(__file__))
-    dirs = [here, join(here, "virtualenv_support")]
-    if os.path.splitext(os.path.dirname(__file__))[0] != "virtualenv":
-        # Probably some boot script; just in case virtualenv is installed...
+@contextlib.contextmanager
+def virtualenv_support_dirs():
+    """Context manager yielding either [virtualenv_support_dir] or []"""
+
+    # normal filesystem installation
+    if os.path.isdir(join(HERE, "virtualenv_support")):
+        yield [join(HERE, "virtualenv_support")]
+    elif IS_ZIPAPP:
+        tmpdir = tempfile.mkdtemp()
```

Comment:
worth writing a comment here that we provide these files on the fly, so we'll extract it and clean it up afterwards 👍

That being said seems fairly expensive to keep extracting this every time, worth investigating if:
1. Can't we provide it as a zip path?
2. Can we cache it, into e.g. userdir?

Reply:
premature optimization (~650μs per call on my VM)

also think it's fairly obvious what's happening based on the code (not warranting a comment) -- let me know!